res/res_pjsip_sdp_rtp.c

```diff
@@ -1963,20 +2019,41 @@ static int create_outgoing_sdp_stream(struct ast_sip_session *session, struct as
 			if (!(noncodec & index)) {
 				continue;
 			}
-			rtp_code = ast_rtp_codecs_payload_code(
-				ast_rtp_instance_get_codecs(session_media->rtp), 0, NULL, index);
-			if (rtp_code == -1) {
-				continue;
-			}
 
-			if ((attr = generate_rtpmap_attr(session, media, pool, rtp_code, 0, NULL, index))) {
-				media->attr[media->attr_count++] = attr;
-			}
 
-			if (index == AST_RTP_DTMF) {
-				snprintf(tmp, sizeof(tmp), "%d 0-16", rtp_code);
-				attr = pjmedia_sdp_attr_create(pool, "fmtp", pj_cstr(&stmp, tmp));
-				media->attr[media->attr_count++] = attr;
+			if (index != AST_RTP_DTMF) {
+				rtp_code = ast_rtp_codecs_payload_code(
+								ast_rtp_instance_get_codecs(session_media->rtp), 0, NULL, index);
+				if (rtp_code == -1) {
+					continue;
+				}
+				if ((attr = generate_rtpmap_attr(session, media, pool, rtp_code, 0, NULL, index))) {
+					media->attr[media->attr_count++] = attr;
+				}
+			}
+			else {
+				/*
+				 * Walk through the possible bitrates for the 2833/4733 digits and generate the rtpmap
+				 * attributes.
+				 */
+				for (int i = 0; i < AST_RTP_NUM_DTMF_SAMPLERATES; i++) {
+					if (sampleRates[i] != 0) {
```

Comment:
If you change this to a vector which shrinks/grows it'll handle all sample rates without you caring.